Shared/Notifications/NotificationsManager.swift

```diff
@@ -0,0 +1,211 @@
+                let notifInfo = request.content.userInfo
+                guard let menuItemName = notifInfo[self.notifContentMenuItemKey] as? String else { continue }
+                if var scheduledNotifications = self.scheduledNotifications[menuItemName] {
+                    scheduledNotifications.append(request)
+                    self.scheduledNotifications[menuItemName] = scheduledNotifications
+                } else {
+                    self.scheduledNotifications[menuItemName] = [request]
+                }
+            }
+        })
+    }
+    
+    /// Verifies all notifications for favorite menu items and updates them according to the most recent menu data
+    func verifyCurrentNotifications() {
+        populateScheduledNotifications()
+        updateNotifications(menuItemNames: Defaults[\.favoriteMenuItems])
+    }
+    
+    /// Schedule a notification for a menu item being served at any of the eateries in eateryDisplayNames, to be triggered on date
+    private func setUpNotification(menuItemName: String, eateryDisplayNames: [String], on date: Date, eventName: CampusEatery.EventName) {
+        guard date > Date(), !eateryDisplayNames.isEmpty else { return }
+        
+        let notifContent = UNMutableNotificationContent()
+        notifContent.title = self.favoriteMenuItemNotifTitle
+        var multipleServingsPostfix: String
+        switch eateryDisplayNames.count {
+        case 1:
+            multipleServingsPostfix = ""
+        case 2:
+            multipleServingsPostfix = " and \(eateryDisplayNames[1])"
+        case 3:
+            multipleServingsPostfix = ", \(eateryDisplayNames[1]), and 1 other"
+        default:
+            multipleServingsPostfix = ", \(eateryDisplayNames[1]), and \(eateryDisplayNames.count - 2) others"
+        }
+        notifContent.body = "\(menuItemName) is being served at \(eateryDisplayNames[0])" + multipleServingsPostfix
+        notifContent.sound = UNNotificationSound.default()
+        notifContent.userInfo = [
+            notifContentDateKey : date,
+            notifContentEateriesKey : eateryDisplayNames,
+            notifContentEventNameKey: eventName,
+            notifContentMenuItemKey : menuItemName
+        ]
+
+        let notifDateComponents = Calendar.current.dateComponents([.second, .minute, .hour, .day, .month, .year], from: date)
+        let notifTrigger = UNCalendarNotificationTrigger(dateMatching: notifDateComponents, repeats: false)
+        let notifRequest = UNNotificationRequest(identifier: UUID().uuidString, content: notifContent, trigger: notifTrigger)
+        notificationCenter.add(notifRequest, withCompletionHandler: nil)
+        
+        var newMenuItemNotifRequests: [UNNotificationRequest]
+        if let menuItemNotifRequests = scheduledNotifications[menuItemName] {
+            newMenuItemNotifRequests = menuItemNotifRequests
+        } else {
+            newMenuItemNotifRequests = [UNNotificationRequest]()
+        }
```

Comment:
could we do: `var newMenuItemNotifRequests: [UNNotificationRequest] = scheduledNotifications[menuItemName] ?? []`